BLSQ/iaso
# IA-3560 Org Unit Type restrictions in `UserRole` (backend)

Allow to edit/create org units based on a set of org unit types.

Related JIRA tickets : [IA-3560](https://bluesquare.atlassian.net/browse/IA-3560)

## Changes

- implement org unit type restrictions in `UserRole` model
- implement org unit type restrictions in `UserRole` API
- add a `user_roles_editable_org_unit_type_ids` field to the Profile API

## How to test

See unit tests.

Use the DRF UI to see the read payload: `http://localhost:8081/api/userroles/`

![1](https://github.com/user-attachments/assets/7016b3df-5ca0-4a22-bac5-08dde15689ea)

![2](https://github.com/user-attachments/assets/7d32f8f1-c427-442e-951c-2a22bf51d1f9)

![field](https://github.com/user-attachments/assets/b39f29ac-d306-4951-ac56-576e8a10515c)

## Notes

This PR is inspired by #1658 (but do some things differently)

The scope of this PR is limited to the `UserRole` model.


[IA-3560]: https://bluesquare.atlassian.net/browse/IA-3560?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
